### PR TITLE
Added authorization and authentification.

### PR DESCRIPTION
**Pull request by vkhmara**

### CODE REVIEW

File: task8/ServletsOnMaven/src/main/java/Posts/JSONDecorator.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package ServletsOnPosts;
+package Posts;
```
Comment (smiakchilo) [inline task8/ServletsOnMaven/src/main/java/Posts/JSONDecorator.java:1]: Packages are named in all-small letters in Java

File: task8/ServletsOnMaven/src/main/java/Services/UserService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public static class DifferencePasswords extends Exception {
+
+    }
+
+    public static void checkForUser(User user, DataSource dataSource) throws NoUser, DifferencePasswords {
+        String password = WorkWithDB.getPassword(user.username, dataSource);
+        if (password == null) {
+            throw new NoUser();
```
Comment (smiakchilo) [inline task8/ServletsOnMaven/src/main/java/Services/UserService.java:27]: By convention, exception classes are always named ending with `...Exception`

File: task8/ServletsOnMaven/src/main/java/ServletsOnPosts/SignupServlet.java [hunk 3]
```diff
@@ -12,28 +13,35 @@
...
+        }
+        catch (UserService.NoUser e) {
+            writer.write(JSONDecorator.toJson("No such user"));
             return;
         }
-        String passwordUser = WorkWithDB.getPassword(username, dataSource);
-        resp.setContentType("application/json");
-        resp.getWriter().write(JSONDecorator.toJson(passwordUser));
+        catch (UserService.DifferencePasswords ignored) {}
+        writer.write(JSONDecorator.toJson("Yes such user"));
```
Comment (smiakchilo) [inline task8/ServletsOnMaven/src/main/java/ServletsOnPosts/SignupServlet.java:35]: This is not English))

File: task8/ServletsOnMaven/src/main/java/ServletsOnPosts/SignupServlet.java [hunk 4]
```diff
@@ -12,28 +13,35 @@
...
+            UserService.checkForUser(new User(username, ""), dataSource);
+        }
+        catch (UserService.NoUser e) {
+            writer.write(JSONDecorator.toJson("No such user"));
             return;
         }
-        String passwordUser = WorkWithDB.getPassword(username, dataSource);
-        resp.setContentType("application/json");
-        resp.getWriter().write(JSONDecorator.toJson(passwordUser));
+        catch (UserService.DifferencePasswords ignored) {}
```
Comment (smiakchilo) [inline task8/ServletsOnMaven/src/main/java/ServletsOnPosts/SignupServlet.java:34]: Exceptions should never be ignored, without at least a comment telling why

File: task8/ServletsOnMaven/src/main/java/ServletsOnPosts/LoginServlet.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        User postRequest = JSONDecorator.getGson().fromJson(body, User.class);
+        try {
+            UserService.checkForUser(postRequest, dataSource);
+        }
+        catch (UserService.NoUser noUser) {
+            writer.write(JSONDecorator.toJson("No such user"));
+            return;
+        }
+        catch (UserService.DifferencePasswords differencePasswords) {
+            writer.write(JSONDecorator.toJson("Passwords are not equal"));
```
Comment (smiakchilo) [inline task8/ServletsOnMaven/src/main/java/ServletsOnPosts/LoginServlet.java:37]: Are you reporting that the user-input password does not match the one stored in DB?
If so, this is still a security flaw. Nowadays, most ofter the server replies "No such user of a bad password". We don;t give a spare confirmation that such user exists thus welcoming an intruder to try and brute-froce a password

**Review by smiakchilo — COMMENTED** [review]
BE part checked 29/05/2020

**Comment by vkhmara** [comment]
Ok, thanks for your all remarks, I found out some new information) But the time is over and I can't pull any requests. Can I merge this branches?

**Comment by smiakchilo** [comment]
Yeah, go ahead) От: vkhmaraОтправлено: пятница, 29 мая 2020 г. в 09:46Кому: vkhmara/upreposКопия: Stepan Miakchilo; Review requestedТема: Re: [vkhmara/uprepos] Added authorization and authentification. (#14) Ok, thanks for your all remarks, I found out some new information) But the time is over and I can't pull any requests. Can I merge this branches?—You are receiving this because your review was requested.Reply to this email directly, view it on GitHub, or unsubscribe. 
CONFIDENTIALITY
NOTICE: This email and files attached to it are confidential. If you
are not the intended recipient you are hereby notified that using,
copying, distributing or taking any action in reliance on the contents of this information is strictly prohibited. If you have
received this email in error please notify the sender and delete this
email.

**Pull request merged**